hal/shared/platform_ncp.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2018 Particle Industries, Inc.  All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "module_info.h"
+#include "ota_flash_hal.h"
+#include "hal_platform.h"
+
+enum PlatformNCPManufacturer {
+    PLATFORM_NCP_MANUFACTURER_UNKNOWN,
+    PLATFORM_NCP_MANUFACTURER_ESPRESSIF = 1,
+    PLATFORM_NCP_MANUFACTURER_UBLOX = 2,
+    PLATFORM_NCP_MANUFACTURER_QUECTEL = 3,
+    PLATFORM_NCP_MANUFACTURER_BROADCOM = 4
+};
+
+#define PLATFORM_NCP_IDENTIFIER(mf,t) (t|(mf<<5))
+
+enum PlatformNCPIdentifier {
+    PLATFORM_NCP_UNKNOWN = -1,
+    PLATFORM_NCP_NONE = 0,
+    PLATFORM_NCP_ESP32 = PLATFORM_NCP_IDENTIFIER(PLATFORM_NCP_MANUFACTURER_ESPRESSIF, 1),
+    PLATFORM_NCP_SARA_U201 = PLATFORM_NCP_IDENTIFIER(PLATFORM_NCP_MANUFACTURER_UBLOX, 2),
+    PLATFORM_NCP_SARA_G350 = PLATFORM_NCP_IDENTIFIER(PLATFORM_NCP_MANUFACTURER_UBLOX, 3),
+    PLATFORM_NCP_SARA_R410 = PLATFORM_NCP_IDENTIFIER(PLATFORM_NCP_MANUFACTURER_UBLOX, 4),
+    PLATFORM_NCP_QUECTEL_BG96 = PLATFORM_NCP_IDENTIFIER(PLATFORM_NCP_MANUFACTURER_QUECTEL, 5),
+    PLATFORM_NCP_QUECTEL_EG91 = PLATFORM_NCP_IDENTIFIER(PLATFORM_NCP_MANUFACTURER_QUECTEL, 6),
+    PLATFORM_NCP_SARA_U260 = PLATFORM_NCP_IDENTIFIER(PLATFORM_NCP_MANUFACTURER_UBLOX, 7),
```

Comment:
Shouldn't these two be `(PLATFORM_NCP_MANUFACTURER_UBLOX, 5)` and `(PLATFORM_NCP_MANUFACTURER_UBLOX, 6)` to continue the ids from the other ublox NCP types?

Reply:
I just obey the regular:
```
MESH_NCP_ESP32 = MESH_NCP_IDENTIFIER(MESH_NCP_MANUFACTURER_ESPRESSIF, 1),
MESH_NCP_SARA_U201 = MESH_NCP_IDENTIFIER(MESH_NCP_MANUFACTURER_UBLOX, 2),
MESH_NCP_SARA_G350 = MESH_NCP_IDENTIFIER(MESH_NCP_MANUFACTURER_UBLOX, 3),
MESH_NCP_SARA_R410 = MESH_NCP_IDENTIFIER(MESH_NCP_MANUFACTURER_UBLOX, 4),
```